pcp/Platform.c

```diff
@@ -0,0 +1,874 @@
+/*
+htop - linux/Platform.c
+(C) 2014 Hisham H. Muhammad
+(C) 2020-2021 htop dev team
+(C) 2020-2021 Red Hat, Inc.  All Rights Reserved.
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "config.h"
+
+#include <math.h>
+
+#include "BatteryMeter.h"
+#include "ClockMeter.h"
+#include "Compat.h"
+#include "CPUMeter.h"
+#include "DateMeter.h"
+#include "DateTimeMeter.h"
+#include "DiskIOMeter.h"
+#include "HostnameMeter.h"
+#include "LoadAverageMeter.h"
+#include "Macros.h"
+#include "MainPanel.h"
+#include "MemoryMeter.h"
+#include "Meter.h"
+#include "NetworkIOMeter.h"
+#include "Object.h"
+#include "Panel.h"
+#include "PCPProcess.h"
+#include "PCPProcessList.h"
+#include "Platform.h"
+#include "ProcessList.h"
+#include "ProvideCurses.h"
+#include "Settings.h"
+#include "SwapMeter.h"
+#include "SysArchMeter.h"
+#include "TasksMeter.h"
+#include "UptimeMeter.h"
+#include "XUtils.h"
+#include "linux/PressureStallMeter.h"
+#include "linux/ZramMeter.h"
+#include "linux/ZramStats.h"
```

Comment:
This should not be included unconditionally, unless we're really compiling PCP on Linux.

Reply:
I think its OK - we could be looking at a remote Linux host, or an archive from Linux, on some other platform.

Reply:
Mostly cautious because foreign platform headers may include system headers that might not be available for cross-compilation. Like when compiling PCP on FreeBSD. That way this include may reference system headers not available on the actual platform being compiled for (FreeBSD).

Reply:
@BenBE it's a good point.  Fortunately the individual Meters tend to include only Meter.h and then related "core" platform headers for their implementation - and call out to platform-specific code via Platform.h - this is just what we want.  If a platform-specific Meter didn't do this (and couldn't be refactored for some reason), we'd not be able to use that specific Meter for the PCP platform.  That's OK too.